signac/core/h5store.py

```diff
@@ -83,6 +93,9 @@ def _h5set(file, grp, key, value, path=None):
     elif type(value).__module__ == numpy.__name__:
         grp[key] = value
 
+    elif isinstance(value, h5py._hl.dataset.Dataset):
```

Comment:
It would be good to have a corresponding comment above this line, for consistency with other types.